test/unit/indices/proxy_rules_indices_test.rb

```diff
@@ -7,132 +7,65 @@ class ProxyRulesIndicesTest < ActiveSupport::TestCase
+  test 'search for exact word' do
     ThinkingSphinx::Test.rt_run do
       backend_api = FactoryBot.create(:backend_api)
       perform_enqueued_jobs(only: SphinxIndexationWorker) do
-        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/+/')
-        query      = ProxyRuleQuery.new(owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
+        FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/path/foo/to')
+        FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/dont/match/at/all')
+        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/path/to/bar')
 
-        assert_equal [proxy_rule], query.search_for('/+/')
-      end
-    end
-  end
+        results = search_for('bar', owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
 
-  test 'character `,` is indexed' do
-    ThinkingSphinx::Test.rt_run do
-      backend_api = FactoryBot.create(:backend_api)
-      perform_enqueued_jobs(only: SphinxIndexationWorker) do
-        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/,/')
-        query      = ProxyRuleQuery.new(owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
-
-        assert_equal [proxy_rule], query.search_for('/,/')
-      end
-    end
-  end
-
-  test 'character `-` is indexed' do
-    ThinkingSphinx::Test.rt_run do
-      backend_api = FactoryBot.create(:backend_api)
-      perform_enqueued_jobs(only: SphinxIndexationWorker) do
-        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/-/')
-        query      = ProxyRuleQuery.new(owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
-
-        assert_equal [proxy_rule], query.search_for('/-/')
-      end
-    end
-  end
-
-  test 'character `.` is indexed' do
-    ThinkingSphinx::Test.rt_run do
-      backend_api = FactoryBot.create(:backend_api)
-      perform_enqueued_jobs(only: SphinxIndexationWorker) do
-        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/./')
-        query      = ProxyRuleQuery.new(owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
-
-        assert_equal [proxy_rule], query.search_for('/./')
+        assert_equal [proxy_rule.id], results
       end
     end
   end
 
-  test 'character `_` is indexed' do
-    ThinkingSphinx::Test.rt_run do
-      backend_api = FactoryBot.create(:backend_api)
-      perform_enqueued_jobs(only: SphinxIndexationWorker) do
-        proxy_rule = FactoryBot.create(:proxy_rule, owner: backend_api, pattern: '/_/')
-        query      = ProxyRuleQuery.new(owner_type: proxy_rule.owner_type, owner_id: proxy_rule.owner_id)
+  private
 
-        assert_equal [proxy_rule], query.search_for('/_/')
-      end
-    end
+  def search_for(pattern, **options)
+    search_options = ProxyRuleQuery::DEFAULT_SEARCH_OPTIONS.merge options
+    ProxyRule.search(ThinkingSphinx::Query.escape(pattern), search_options).to_a
```

Comment:
This is great, I'm just wondering, can we use `scoped_search`, what we normally use in our code? e.g.
```suggestion
    ProxyRule.scoped_search(by_query: pattern).to_a
```

idk what these options are so perhaps directly accepting this will break everything.

Reply:
What's the difference between `search` and `scoped_search`? I've been looking but I couldn't find any occurrence of `scoped_search` in our code or in `thinking-sphinx` code

Reply:
sorry, it is `scope_search` , used everywhere I believe. I don't think we use plain `search`.

Reply:
But what's the difference between them? The code was already calling `search`, I didn't change that.

`search` is a `ThinkingSphinx` method, but `scope_search` is ours. I read the code and couldn't understand it.

Reply:
The point is that when useing the method directly, you also directly escape the string. So if we change escaping in `serach_scope` that will not automatically reflect in the test. Anyway, not a big deal.